### PR TITLE
[e2e tests] Fix basic spec for multiple environments

### PR DESCRIPTION
**Pull request by adimoldovan**
### Changes proposed in this Pull Request:

`Load the home page` test is failing in default-wpcom and default-pressable because the `powered by` text is different.
Updated the assertion to use a regular expression and accommodate more variants of the text.

### How to test the changes in this Pull Request:

```
pnpm test:e2e:with-env default -g "Load the home page"
```
```
pnpm test:e2e:with-env default-wpcom -g "Load the home page"
```
```
pnpm test:e2e:with-env default-pressable -g "Load the home page"
```

### CODE REVIEW

**Review by Stojdza — APPROVED** [review]
Works as expected for all three envs 👍

File: plugins/woocommerce/tests/e2e-pw/envs/default-pressable/playwright.config.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ config = {
 	retries: 0,
 	projects: [
 		{
-			name: 'default',
+			name: 'default pressable',
```
Comment (Stojdza) [inline plugins/woocommerce/tests/e2e-pw/envs/default-pressable/playwright.config.js:9]: Out of curiosity, why this change?

Comment (adimoldovan) [reply plugins/woocommerce/tests/e2e-pw/envs/default-pressable/playwright.config.js:9]: Looks better in console logs, nothing more :)
